tests/test_toolchain.py

```diff
@@ -74,16 +85,54 @@ def test_create(self):
         assert m_get_toolchain_versions.call_args_list == [
             mock.call('/tmp/android-ndk', mock.ANY)]
         assert m_get_cython_path.call_args_list == [mock.call()]
-        assert m_build_dist_from_args.call_count == 1
+        build_order = [
+            'hostpython3', 'libffi', 'openssl', 'sqlite3', 'python3',
+            'genericndkbuild', 'setuptools', 'six', 'pyjnius', 'android',
+        ]
+        python_modules = []
+        context = mock.ANY
+        project_dir = None
+        assert m_build_recipes.call_args_list == [
+            mock.call(
+                build_order,
+                python_modules,
+                context,
+                project_dir,
+                ignore_project_setup_py=False
+            )
+        ]
+        assert m_run_distribute.call_args_list == [mock.call()]
 
     def test_create_no_sdk_dir(self):
         """
         The `--sdk-dir` is mandatory to `create` a distribution.
         """
         argv = ['toolchain.py', 'create']
-        with mock.patch('sys.argv', argv), pytest.raises(
+        with patch_sys_argv(argv), pytest.raises(
             BuildInterruptingException
         ) as ex_info:
             ToolchainCL()
         assert ex_info.value.message == (
             'Android SDK dir was not specified, exiting.')
+
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="requires python3")
+    def test_recipes(self):
+        """
+        Checks the `recipes` command prints out recipes information without crashing.
+        """
+        argv = ['toolchain.py', 'recipes']
+        with patch_sys_argv(argv), patch_sys_stdout() as m_stdout:
+            ToolchainCL()
+        # check if we have common patterns in the output
+        expected_strings = (
```

Comment:
I like this, very readable in my opinion!

Reply:
Thanks, I almost missed it as it was "resolved" actually :stuck_out_tongue: